hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/ProvidesHoodieConfig.scala

```diff
@@ -81,10 +80,8 @@ trait ProvidesHoodieConfig extends Logging {
         HoodieSyncConfig.META_SYNC_PARTITION_FIELDS.key -> tableConfig.getPartitionFieldProp,
         HoodieSyncConfig.META_SYNC_PARTITION_EXTRACTOR_CLASS.key -> hiveSyncConfig.getStringOrDefault(HoodieSyncConfig.META_SYNC_PARTITION_EXTRACTOR_CLASS),
         HiveSyncConfigHolder.HIVE_SUPPORT_TIMESTAMP_TYPE.key -> hiveSyncConfig.getBoolean(HiveSyncConfigHolder.HIVE_SUPPORT_TIMESTAMP_TYPE).toString,
-        HoodieWriteConfig.UPSERT_PARALLELISM_VALUE.key -> hoodieProps.getString(HoodieWriteConfig.UPSERT_PARALLELISM_VALUE.key, "200"),
```

Comment:
Does this mean that the upsert parallelism cannot be tuned anymore from the SQL statement?  Generally, are the key-value pairs in `Map.apply()` just overrides?

Let's say if upsert parallelism is not set in the combined options, is the default parallelism picked up using the write config?

Reply:
this is a good catch. i took a closer look at `org.apache.hudi.config.HoodieWriteConfig#getUpsertShuffleParallelism` which does not retrieve the default value. There is no reason that we should not use `getIntOrDefault()` in those `getXXXParallelism()` methods

Reply:
These default values in sql was probably added as bad bandaid to make thing pass. We should fix the root cause from config object level.

Reply:
> Does this mean that the upsert parallelism cannot be tuned anymore from the SQL statement? Generally, are the key-value pairs in `Map.apply()` just overrides?

the `combineOptions` method add it from SQLConf, and the properties priority logical is different from the old, `Map.apply()` is highest

Reply:
@yihua @xushiyan 
  
 - Setting here to 200 is clearly a bandaid
 - As Raymond pointed out we should fix this at the root -- where we access these configs (we need to do it for all configs w/ defaults actually)

Reply:
Definitely, we should not hardcode any defaults here.  As long as the configs take effect, I'm good.